packages/blade/src/components/Alert/_decisions/decisions.md

```diff
@@ -0,0 +1,211 @@
+
+> **Note**
+>
+> The following needs work:
+>
+> 1. color on banking theme in high contrast
+> 2. a11y contrast ratio for notice alerts in high contrast
+> 3. the borderless variant mentioned in full bleed layout shouldn't have any borders and border radii
+
+## API
+
+Annotated component with props:
+
+![Alert breakdown](./alert-breakdown.png)
+
+Sample usage:
+
+```jsx
+import { Alert, Link } from '@razorpay/blade';
+
+<Alert
+  title="International Payments Only"
+  description={
+    <>
+      Currently you can only accept payments in international currencies using PayPal. You cannot
+      accept payments in INR (₹) using PayPal. <Link href="https://razorpay.com">Know More</Link>
+    </>
+  }
+  actions={{
+    primary: { text: 'Primary Action', onClick: {() => { /* do something */ }} },
+    secondary: { text: 'Link', onClick: {() => { /* do something */ }}, href: 'https://razorpay.com'},
+  }}
+/>;
+```
+
+### Alert
+
+We'll expose an `Alert` component with the following API:
+
+| Prop          | Type                                                        | Default     | Description                                                                                | Required |
+| ------------- | ----------------------------------------------------------- | ----------- | ------------------------------------------------------------------------------------------ | -------- |
+| description   | `string`, `JSX`                                             | `undefined` | Body content                                                                               | ✅       |
+| title         | `string`                                                    | `undefined` | A brief heading                                                                            |          |
+| isDismissable | `boolean`                                                   | `true`      | Shows a dismiss button                                                                     |          |
+| onDismiss     | `function`                                                  | `undefined` | Callback when the dismiss button is pressed                                                |          |
+| contrast      | `high`, `low`                                               | `low`       | Can be set to `high` for more prominent look _(not related to a11y)_                       |          |
+| isFullWidth   | `boolean`                                                   | `false`     | Spans the entire width of container, otherwise max width is restricted to 584px by default |          |
+| variant       | `info`, `positive`, `notice`, `negative`                    | `info`      | Sets the color tone of entire alert. Icon is set automatically based on variant            |          |
+| isBorderless  | `boolean`                                                   | `false`     | Removes borders and border radii, useful for creating a full bleed layout                  |          |
```

Comment:
ideally since we have isBorderless for a very specific use case, we shall try to restrict people to use it even for non-full width/fleed layouts.

**proposal:** maybe, we shall remove this prop and only have `isFullWidth` which will also control the hiding of border and removing radii. that way we can say that the max-width of the alert will be set to 540px but if you want full width that will always become full bleed and usually full bleed is mostly for cases like banner only.

Reply:
**question:** will isBorderless make the alert absolute positioned, stick at the top, outside the entire dom tree? basically with higher z-index?

Reply:
That would make it hard to build alerts for desktop where they're expected to span full width of container (which might be more than 584px on desktop). Example - the yellow alert on top below:

![image](https://user-images.githubusercontent.com/6682655/182298914-0664496e-99ad-4a67-9282-7b032d9aa506.png)

Reply:
`isBorderless` doesn't change positioning at all, because if we did we would also need to provide a way to set the position for users (top: 0, etc.)

Reply:
Would vote too to keep `isBorderless` independent of any kind of positioning. The decisions for positioning (absolute, sticky, fixed etc.) should be entirely left to the consumer if they want it wrapped inside a container or to the viewport. Even for the full bleed example I can see use cases where consumers might want the Alert popping up from the bottom of the screen, so a `position: fixed; top: 0` wouldn't make sense then.